Symfony/CS/Tokenizer/Token.php

```diff
@@ -99,6 +100,13 @@ public function equals($other)
         $selfPrototype = $this->getPrototype();
 
         foreach ($otherPrototype as $key => $val) {
+            if (!isset($selfPrototype[$key])) {
+                return false;
+            }
+            if ($key === 1 && !$caseSensitive && !strcasecmp($val, $selfPrototype[1])) {
```

Comment:
in your tests `!strcasecmp($val, $selfPrototype[1])` is always true.
please add more test cases